src/components/tl/mlx5/mcast/tl_mlx5_mcast.h

```diff
@@ -468,32 +469,38 @@ static inline ucc_status_t ucc_tl_mlx5_mcast_post_recv_buffers(ucc_tl_mlx5_mcast
     struct ibv_recv_wr *rwr    = comm->call_rwr;
     struct ibv_sge     *sge    = comm->call_rsgs;
     struct pp_packet   *pp     = NULL;
-    int                 count  = comm->params.rx_depth - comm->pending_recv;
     int                 i;
+    int                 j;
+    int                 count;
+    int                 count_per_qp;
 
+    count = comm->params.rx_depth - comm->pending_recv;
     if (count <= comm->params.post_recv_thresh) {
         return UCC_OK;
     }
 
-    for (i = 0; i < count - 1; i++) {
-        if (NULL == (pp = ucc_tl_mlx5_mcast_buf_get_free(comm))) {
-            break;
+    count_per_qp = count / comm->mcast_group_count;
```

Comment:
seems like function doesn't properly handle the case where count_per_qp is not evenly divisible.

Reply:
currently, we do not support that case, and filter those during init.